artifacts/definitions/Windows/Remediation/Quarantine.yaml

```diff
@@ -0,0 +1,359 @@
+      - To update policy, simply rerun the artifact.
+        
+      NOTE:  
+      
+      - Remember DNS resolution may change. It is highly recommended to plan 
+      policy accordingly and not rely on DNS lookups.  
+      - Local IPSec policy can not be applied when Domain IPSec policy
+      is already enforced. Please configure at GPO level in this case.
+
+author: Matt Green - @mgreen27
+
+reference:
+  - https://mgreen27.github.io/posts/2020/07/23/IPSEC.html
+
+required_permissions:
+  - EXECVE
+
+precondition: SELECT OS From info() where OS = 'windows'
+
+parameters:
+  - name: PolicyName
+    default: "VelociraptorQuarantine"
+  - name: RuleLookupTable
+    type: csv 
+    default: |
+        Action,SrcAddr,SrcMask,SrcPort,DstAddr,DstMask,DstPort,Protocol,Mirrored,Description
+        Permit,me,,0,any,,53,udp,yes,DNS
+        Permit,me,,0,any,,53,tcp,yes,DNS TCP
+        Permit,me,,68,any,,67,udp,yes,DHCP
+        Block,any,,,any,,,,yes,All other traffic
+  - name: MessageBox
+    description: |
+        Optional message box notification to send to logged in users. 256 
+        character limit.
+  - name: RemovePolicy
+    type: bool
+    description: Tickbox to remove policy.
+    
+sources:
+    - queries:
+      # If a MessageBox configured truncate to 256 character limit
+      - LET MessageBox <= if(condition=MessageBox,
+                then=parse_string_with_regex(
+                    string=MessageBox,
+                    regex=['^(?P<Message>.{0,255}).*']
+                ).Message)
+                
+      # extract configurable policy from lookuptable
+      - LET configurable_policy <= SELECT 
+                Action,
+                SrcAddr,
+                format(format='%v',args=SrcMask) AS SrcMask,
+                format(format='%v',args=SrcPort) AS SrcPort,
+                DstAddr,
+                format(format='%v',args=DstMask) AS DstMask,
+                format(format='%v',args=DstPort) AS DstPort,
+                Protocol,
+                Mirrored,
+                Description
+            FROM parse_csv(filename=RuleLookupTable, accessor='data') 
+            
+      # extract Velociraptor config for policy 
+      - LET extracted_config <= SELECT * FROM foreach(
+                row=config.server_urls,
+                query={ 
+                    SELECT 
+                        'Permit' AS Action,
+                        'me' AS SrcAddr,
+                        '' As SrcMask,
+                        '0' AS SrcPort,
+                        split(
+                            string=regex_replace(
+                                source=_value,
+                                re='^https://(.+)/',
+                                replace='$1'),
+                            sep=':'
+                                )[0] as DstAddr,
+                                '' As DstMask,
+                        if(condition=split(string=regex_replace(
```

Comment:
You can do something like
```
parse_string_with_regex(regex='^https?://[^:/]+(:(?P<Port>[0-9]*))?/', string=X).Port AS DstPort
```

I would leave the if statement until later to substitute with "443"

Reply:
ran this then updated logic further down the chain to cover https + http use case.